### PR TITLE
add benchmarks for accumulate and cumsum

### PR DESCRIPTION
**Pull request by simonbyrne**
For testing https://github.com/JuliaLang/julia/pull/25766

### CODE REVIEW

**Comment by simonbyrne** [comment]
Not sure what the error is: @ararslan do we need to make changes here to get this working on 0.7?

File: src/array/ArrayBenchmarks.jl [hunk 1]
```diff
@@ -339,4 +339,33 @@ for x in (x_range, collect(x_range), Int16.(x_range), Float64.(x_range), Float32
...
+resaint = similar(bint)
+
+g["accumulate!", "Float64"] = @benchmarkable accumulate!(+, $resafloat, $afloat)
+g["accumulate!", "Int"] = @benchmarkable accumulate!(+, $resaint, $aint)
+
+g["cumsum!", "Float64"] = @benchmarkable cumsum!($resafloat, $afloat)
+g["cumsum!", "Int"] = @benchmarkable cumsum!($resaint, $aint)
+
+mfloat = samerand(10^3,10^3)
+g["cumsum", "Float64", "dim1"] = @benchmarkable cumsum($mfloat, dims=1)
```
Comment (ararslan) [inline src/array/ArrayBenchmarks.jl]: I think this will need to be `Compat.cumsum` (similarly with the others) for the `dims` keyword argument.

**Comment by ararslan** [comment]
I have absolutely no idea what that error is about.

**Comment by ararslan** [comment]
Just FYI I won't be able to retune benchmarks and rerun Nanosoldier on your PR until MIT is finished with their on-site stuff and the machines are turned back on.

File: src/array/ArrayBenchmarks.jl [hunk 2]
```diff
@@ -339,4 +339,33 @@ for x in (x_range, collect(x_range), Int16.(x_range), Float64.(x_range), Float32
...
+g["cumsum", "Int"] = @benchmarkable cumsum($aint)
+
+resafloat = similar(afloat)
+resaint = similar(aint)
+
+g["accumulate!", "Float64"] = @benchmarkable accumulate!(+, $resafloat, $afloat)
+g["accumulate!", "Int"] = @benchmarkable accumulate!(+, $resaint, $aint)
+
+g["cumsum!", "Float64"] = @benchmarkable cumsum!($resafloat, $afloat)
+g["cumsum!", "Int"] = @benchmarkable cumsum!($resaint, $aint)
```
Comment (jw3126) [inline src/array/ArrayBenchmarks.jl]: There are overlapping benchmarks [here](https://github.com/JuliaCI/BaseBenchmarks.jl/blob/f6dfcc0c8b45d6d3c43f94548994494ae2c69427/src/linalg/LinAlgBenchmarks.jl#L106), maybe they should be removed.

**Pull request merged**